src/Symfony/Component/Validator/Constraints/BackedEnumValueValidator.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * EnumValidator validates that the value is one of the expected values.
+ *
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
+ */
+class BackedEnumValueValidator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof BackedEnumValue) {
+            throw new UnexpectedTypeException($constraint, BackedEnumValue::class);
+        }
+
+        if (!is_a($constraint->type, \BackedEnum::class, true)) {
+            throw new ConstraintDefinitionException('The "type" must be a \BackedEnum.');
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        $backingType = (string) (new \ReflectionEnum($constraint->type))->getBackingType();
+        if ($this->getNormalizedType($value) !== $backingType) {
```

Comment:
Does that mean, we would not allow numeric strings on integer-backed enums? Do we want that level of strictness?